LangChainQuestionGenerator.py

```diff
@@ -252,21 +257,29 @@ def retrieveLangChainQuestions(config, videoData=None, overwrite=False):
     return questionData
 
 
-def makeRetriever(transcript, embeddings) -> Chroma:
+def makeRetriever(transcript, embeddings, collectionName) -> Chroma:
     """
-    Creates a retriever object using the given transcript and embeddings.
+    Creates a retriever object using the given transcript, embeddings, and collection name.
 
     Args:
         transcript (str): The transcript to be used for creating the retriever.
         embeddings: The embeddings to be used for creating the retriever.
+        collectionName (str): The name of the collection.
 
     Returns:
         Chroma: The retriever object.
 
+    The collectionName matches the video name or parent folder name.
+    This is done to ensure that each transcript is associated with a unique retriever object.
+    So any data generated from the retriever can be associated with the correct video or parent folder.
     """
+    
+    collectionName = collectionName.replace(" ", "_")
     transcript = getMetadata(transcript)
     loader = DataFrameLoader(transcript, page_content_column="Combined Lines")
-    vectorstore = Chroma.from_documents(documents=loader.load(), embedding=embeddings)
+    vectorstore = Chroma.from_documents(
+        documents=loader.load(), embedding=embeddings, collection_name=collectionName
+    )
```

Comment:
So call it a Novice Question.... opportunity for me learn

VectorStore is used to store `embedding` info which is a some collection of data in number format. So I know currently the info we have is not a lot. So the info seems to be stored in MemCache of the machine. I have heard vector Databases that is used for the AI/GAI application. what is the use case where VectorStore Databases are used?